src/subscription_manager/cli_command/register.py

```diff
@@ -329,39 +329,45 @@ def _prompt_for_environment(self):
         readline.clear_history()
         return environment or self._prompt_for_environment()
 
-    def _process_environments(self, admin_cp, owner_key, options):
+    def _process_environments(self, admin_cp, owner_key):
         """
         Confirms that environment(s) have been chosen if they are supported
         and a choice needs to be made
         """
         supported_resources = get_supported_resources()
         supports_environments = 'environments' in supported_resources
 
-        if not supports_environments and self.options.environments is not None:
-            system_exit(os.EX_UNAVAILABLE, _("Error: Server does not support environments."))
+        if not supports_environments:
+            if self.options.environments is not None:
+                system_exit(os.EX_UNAVAILABLE, _("Error: Server does not support environments."))
+            return None
 
-        if supports_environments:
-            all_env_list = admin_cp.getEnvironmentList(owner_key)
-            if self.options.environments:
-                environments = options.environments
-            else:
-                # If there aren't any environments, don't prompt for one
-                if not all_env_list:
-                    return None
-
-                # If the envronment list is len 1, pick that environment
-                if len(all_env_list) == 1:
-                    log.debug("Using the only available environment: \"{name}\"".format(name=all_env_list[0]["name"]))
-                    return all_env_list[0]['id']
-
-                env_name_list = [env['name'] for env in all_env_list]
-                print(_('Hint: Organization "{key}" contains following environments: {list}').format(
-                      key=owner_key, list=", ".join(env_name_list)))
-                environments = self._prompt_for_environment()
-                if not self.cp.has_capability(MULTI_ENV) and ',' in environments:
-                    system_exit(os.EX_USAGE, _("The entitlement server does not allow multiple environments"))
-
-            return check_set_environment_names(all_env_list, environments)
+        # We have an activation key, so don't need to fill/check the bits
+        # related to environments, as they are part of the activation key
+        if self.options.activation_keys:
+            return None
+
+        all_env_list = admin_cp.getEnvironmentList(owner_key)
+        if self.options.environments:
+            environments = self.options.environments
+        else:
+            # If there aren't any environments, don't prompt for one
+            if not all_env_list:
+                return None
+
+            # If the envronment list is len 1, pick that environment
```

Comment:
Please fix typo 'envronment'.

Reply:
it is: it is not related to my changes

Reply:
What about adding another commit to this PR?

Reply:
I'd rather introduce tools like codespell for the whole codebase and fix all the typos/issues at once, rather than ad-hoc approach for every small typo...

Reply:
I'm 100% for fixing all typos. Great idea! Automated check of typos as CI also good idea.

Can you just fix this typo, when I asked for that and I said please? Thanks.

Reply:
It's out of scope of this PR, so no. Thanks for asking though.